packages/search/README.md

```diff
@@ -55,6 +55,117 @@ const { agoSearch } = require('@esri/hub-search')
+
+Examples:
+```
+// corresponds to (tag: TAG_ONE OR tag: TAG_TWO)
+const filterOne = { tag: { bool: "OR", value: ["TAG_ONE", "TAG_TWO"] } };
+
+// corresponds to (tag: TAG_ONE AND tag: TAG_TWO)
+const filterTwo = { tag: { bool: "AND", value: ["TAG_ONE", "TAG_TWO"] } };
+
+// corresponds to (-tag: TAG_ONE AND -tag: TAG_TWO)
+const filterThree = { tag: { bool: "NOT", value: ["TAG_ONE", "TAG_TWO"] } };
+```
+
+Through these three options, large filters can be constructed. As a default, items of type `code attachment` are excluded from search.
+
+Example:
+```
+// corresponds to `(water) AND (owner: me OR owner: you) AND (created: [1609459200000 TO 1612137600000]) AND (modified: [1609459200000 TO 1612137600000]) AND (-title: "a title" AND -title: "b title") AND (typekeywords: "a type keyword") AND (tags: "tag 1" OR tags: "tag 2" OR tags: "tag 3") AND (type: "Feature Layer" OR type: "Table" OR type: "CSV") AND (access: private) AND (culture: en OR culture: de) AND (categories: "category one" AND categories: "category 2" AND categories: "category three") AND (-type: "code attachment")`
+
+const filters = {
+  terms: "water",
+  owner: ["me", "you"],
+  created: { from: 1609459200000, to: 1612137600000 },
+  modified: { from: 1609459200000, to: 1612137600000 },
+  title: { bool: "NOT", value: ["a title", "b title"] },
+  typekeywords: "a type keyword",
+  tags: ["tag 1", "tag 2", "tag 3"],
+  type: { value: ["Feature Layer", "Table", "CSV"] },
+  access: "private",
+  culture: ["en", "de"],
+  categories: {
+    value: ["category one", "category 2", "category three"],
+    bool: "AND"
+  }
+};
+```
+
+The `options` property provides several familar options for sorting and paging data, as well as requesting aggregations. These are specified by the `IContentSearchOptions` interface. Importantly, a user can also provide an `UserSession` instance as part of the options for authenticated searches. For the `searchContent` function, the client must provide the UserSession instance, `portal` sharing url, and `isPortal` flag, as needed, for each invocation. This is because it is a standalone function with no default options to reference. By contrast, the `ContentSearchService` will use the options it was configured with as a default if the relevant option is not provided.
+
+#### Content Search Results
+The results of a content search could include:
+  - `results`: a list of content returned from the search
+  - `count`: the number of results returned from the request
+  - `total`: the total number of results that will be returned from multiple paginated searches
+  - `query`: the stringified query used to complete the search,
+  - `aggregations`: any aggregations returned
+  - `hasNext`: a boolean indicating if there are more pages of results
+  - `next`: a function that, when invoked, will fetch the next page of results
+
+Importantly, one can optionally provide a UserSession instance to the `next` function invocation to be used for authenticated searches
```

Comment:
Not quite following this... what scenario would you issue a search w/o a `session`, and then need/want to add it on the `.next` invocation? I assume `.next` is a closure over some state, and if a `session` was provided on the initial call we could hold that in the closure? Of course this gives me the memory-leak-willies, b/c we're holding class instances in closures of indeterminate lifetimes... but @mikeringrose  tells me this is "ok", so I'm trying to get over that ;)

Reply:
I originally included it in an effort to keep it as "stateless" as possible, ie. you can always pass in authentication with each invocation if you want to.

I'm not an expert in Ember and its memory management but given what is currently in the UI, I don't believe we would be at significant risk. It looks like old `searchResults`, which would include the `next` function, should be dereferenced, but happy to chat more about it.

```
async fetchMore () {
  const currentResults = this.model.searchResults.results;
  const res = await this.model.searchResults.next();
  res.results = [...currentResults, ...res.results];
  this.set('model.searchResults', res);
},
```